Colorlib: add lighten/darken functions

petzku opened the pull request.

petzku committed:
Colorlib: add lighten/darken functions

petzku committed:
Add minor documentation

petzku committed:
Add documentation for lightening functions

petzku committed:
Add lamenting note about lighten being slightly broken

The0x539 commented:
Either provide a `darken_complex` function or document its absence.

The0x539 commented:
`darken` is not defined as an "inverse" of `lighten`. This seems like a silly oversight.
`darken_complex` should either be provided or have its absence documented.

petzku committed:
color: Remove `darken` function

Functionally useless as `lighten` does the exact same thing.
Negative lightening is just darkening.
